linker/Linker.Steps/MarkStep.cs

```diff
@@ -1587,5 +1598,116 @@ protected virtual void MarkInterfaceImplementation (TypeDefinition type, Interfa
 			MarkCustomAttributes (iface);
 			MarkType (iface.InterfaceType);
 		}
+
+
+		void MarkSomethingUsedViaReflection (string reflectionMethod, Action<Collection<Instruction>, string, TypeDefinition, BindingFlags> markMethod, Collection<Instruction> instructions)
+		{
+			for (var i = 0; i < instructions.Count; i++) {
+				var instruction = instructions [i];
+				if (instruction.OpCode == OpCodes.Call || instruction.OpCode == OpCodes.Callvirt) {
+					var methodBeingCalled = instruction.Operand as MethodReference;
+					if (methodBeingCalled != null && methodBeingCalled.DeclaringType.Name == "Type" && methodBeingCalled.DeclaringType.Namespace == "System")
+					{
+						if (methodBeingCalled.Name == reflectionMethod) {
+							_context.Tracer.Push ($"Reflection-{methodBeingCalled}");
+							var nameOfThingUsedViaReflection = OperandOfNearestInstructionBefore<string> (i, OpCodes.Ldstr, instructions);
+							var bindingFlags = (BindingFlags) OperandOfNearestInstructionBefore<sbyte> (i, OpCodes.Ldc_I4_S, instructions);
```

Comment:
You cannot cast it to BindingFlags as it can be a different type, e.g mscorlib and netcore or version mismatch etc

Reply:
I'm unclear about what you mean here, can you elaborate?

Reply:
You are casting the value to BindingFlags type which is used when linker is compiled and that's wrong. Imagine you are linking 2.0 mscorlib but BindingFlags type used for linker compilation is 4.0 mscorlib (there are more variations of this)

Reply:
I'd prefer to copy the BindingFlags enum into linker for this purposes instead of reusing System.Reflection

Reply:
Ok, that makes sense. So does that mean the linker cannot assume a given `sbyte` value loaded on the stack corresponds to a given value in the `BindingsFlags` enum?

Reply:
Why should we copy the `BindingFlags` enum into the linker code? We would be grabbing the value at a specific point in time. I'm not sure this would help with a possible future change to the enum values.